src/Lean/Elab/App.lean

```diff
@@ -1035,7 +1035,7 @@ private def resolveLValAux (e : Expr) (eType : Expr) (lval : LVal) : TermElabM L
   if eType.isForall then
     match lval with
     | LVal.fieldName _ fieldName _ _ =>
-      let fullName := `Function ++ fieldName
+      let fullName := `Function ++ .mkSimple fieldName
```

Comment:
In general, this transformation is not always correct, because `Name.append` propagates macro scopes from the left (but not right) argument to the result. But I think it's fine here.